docs/spec.md

```diff
@@ -702,195 +706,401 @@ expressions must not be preceded by whitespace before the `[` character,
+before the opening brace `[` in an index expression or slice
+expression. However, it is allowed _within_ the grouping expression,
+index expression, and slice expression, even if the expression is an
+element of a list such as an argument to a function call.
+
+Assignments, inferred variable declarations, return statements and the
+the expression inside an index expression `[ … ]` _can_ have whitespace
+around their binary operators. The whitespace around the operators is
+optional, but it is often used to improve the readability of the code,
+for example:
+
+```evy
+x := 5 + 3
+x = 7 - 2
+
+arr := [1 2 3]
+arr[3 - 2] = 10
+
+func fn:num
+    return 7 + 1
+end
+
+print x arr (fn)
+```
+
+More formally, horizontal whitespace `WS` between tokens or terminals as
+defined in the grammar is ignored and can be used freely with the
+addition of the following rules:
+
+1. `WS` is not allowed around dot `.` in dot expressions.
+2. `WS` is not allowed around before the `[` in index or slice expressions.
+3. `WS` is not allowed following the unary operators `-` and `!`.
+4. `WS` is not allowed within arguments to a function call
+5. `WS` is not allowed within elements of an array literal
+6. `WS` is not allowed within the values of a map literal's key-value pairs.
+7. `WS` is allowed within any grouping expression `(` … `)`.
+8. `WS` is allowed within an index expression `[` … `]`.
+9. `WS` is allowed within a slice expression `[` … `:` … `]`.
+
+Here are some examples of incorrect uses of horizontal whitespace, along
+with their correct uses.
+
+Invalid:
+
+```
+print - 5
+len "a" + "b"
+
+arr := [1 + 1]
+arr [0] = 3 + 2
+print 2 + arr [0]
+
+map := {address: "10 Downing" + "Street"}
+map.  address = "221B Baker Street"
+
+print len map
+```
+
+Valid:
+
+```evy
+print -5
+len "a"+"b"
+
+arr := [1+1]
+arr[0] = 3 + 2
+print 2+arr[0]
+
+map := {address:"10 Downing"+"Street"}
+map.address = "221B Baker Street"
+
+print (len map)
+```
 
 ## Functions
 
-A function declaration binds an identifier, the function name, to a
-function. As part of the function declaration, the function signature
-defines the number, order and types of input parameters as well as the
-result or return type of the function. If the return type is left out
+_Functions_ are blocks of code that are used to perform a specific task.
+They are often used to encapsulate code that is used repeatedly, so
+that it can be called from different parts of a program.
+
+A _function definition_ binds an identifier, the function name, to a
+function. As part of the function definition, the _function signature_
+declares the number, order and types of input parameters as well as the
+result or return type of the function. If the return type is left out,
 the function does not return a value.
 
-    func is_valid:bool text:string cap:num
-        return (len text) <= cap
-    end
+For example, the following code defines a function called `is_valid`
```

Comment:
You camel-cased the function in the code, but forgot to update this instance of `is_valid`.

Reply:
fixed, thank you!